exercises/palindrome-products/.meta/template.j2

```diff
@@ -0,0 +1,49 @@
+"""	
+Notes regarding the implementation of smallest and largest:
+
+Both functions must take two keyword arguments:
+    max_factor -- int
+    min_factor -- int, default 0
+
+Their return value must be a tuple (value, factors) where value is the
+palindrome itself, and factors is an iterable containing both factors of the
+palindrome in arbitrary order.
+"""
+
+{%- import "generator_macros.j2" as macros with context -%}
+
+{%- macro value_factor_unpacking(case) -%}
+    {%- set input = case["input"] -%}
+    value, factors = {{ case["property"] }}(min_factor={{ input["min"] }}, max_factor={{ input["max"] }})
+{%- endmacro -%}
+
+{%- macro test_case(case) -%}
+    {%- set expected = case["expected"] -%}
+    def test_{{ case["description"] | to_snake }}(self):
+    {%- if case is error_case %}
+        with self.assertRaisesWithMessage(ValueError):
+            {{ value_factor_unpacking(case) }}
+    {%- else %}
+        {{ value_factor_unpacking(case) }}
+        {%- if expected["value"] is none %}
+        self.assertIsNone(value)
+        {%- else %}
+        self.assertEqual(value, {{ expected["value"] }})
+        {%- endif %}
+        self.assertFactorsEqual(factors, {{ expected["factors"] }})
+    {%- endif %}
+{% endmacro %}
+
+{{ macros.header() }}
+
+class {{ exercise | camel_case }}Test(unittest.TestCase):
+    {% for case in cases -%}
+    {{ test_case(case) }}
+    {% endfor -%}    
+
+    {{ macros.utility() }}
+
+    def assertFactorsEqual(self, actual, expected):
+        self.assertEqual(set(map(frozenset, actual)), set(map(frozenset, expected)))
+
+{{ macros.footer() }}
```

Comment:
```suggestion
{{ macros.footer(has_error_case) }}
```

Reply:
@cmccandless 

Yeah, I saw that parameter for footer, I even started with that, then changed to what you see now. Thing is, in the proposed way definition of method `assertFactorsEqual` doesn't land under section `# Utility functions` - it's above it. I wanted to make as little changes to the original `palindrome_products_test.py` file as possible.

If you're ok with this change, I am also, and will fix it.

Reply:
Moving the exercise-specific utility function above the global one(s) is fine.

Reply:
See #2084 ; providing the explicit argument is no longer required.

Reply:
Ok, changed.
If you don't mind I also fixed here the humming test template so it doesn't render utilities twice - spotted in your PR #2084